### PR TITLE
Add support for Gen3/Gen4 loaders for Xen DomD

### PR DESCRIPTION
**Pull request by firscity**

### CODE REVIEW

**Comment by sa-kib** [comment]
This fixup should also be applied:
https://github.com/sa-kib/u-boot-rcar/commit/976ff196b33e6d866dcccc73ab3c0ca40841a2c9

File: drivers/ufs/ufs.c [hunk 1]
```diff
@@ -1442,8 +1456,42 @@ static void prepare_prdt_table(struct ufs_hba *hba, struct scsi_cmd *pccb)
...
+	if (sense_buffer &&
+	    ufshcd_get_rsp_upiu_data_seg_len(ucd_rsp_ptr)) {
+		int len_to_copy;
+
+		len = be16_to_cpu(ucd_rsp_ptr->sr.sense_data_len);
+		len_to_copy = min_t(int, RESPONSE_UPIU_SENSE_DATA_LENGTH, len);
+
+		memcpy(sense_buffer, ucd_rsp_ptr->sr.sense_data, len_to_copy);
+	}
+}
```
Comment (sa-kib) [inline drivers/ufs/ufs.c]: Code of `ufshcd_get_rsp_upiu_data_seg_len()` & `ufshcd_copy_sense_data()` routines not needed, remove it from this patch

File: board/renesas/rcar3-xen/rcar3-xen.c [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+	xen_mem_map[i].attrs = (PTE_BLOCK_MEMTYPE(MT_NORMAL) |
+				PTE_BLOCK_INNER_SHARE);
+	i++;
+
+	xen_mem_map[i].virt = 0xC0000000UL;
+	xen_mem_map[i].phys = 0xC0000000UL;
+	xen_mem_map[i].size = 0x40000000UL;
+	xen_mem_map[i].attrs = PTE_BLOCK_MEMTYPE(MT_DEVICE_NGNRNE) |
+			       PTE_BLOCK_NON_SHARE | PTE_BLOCK_PXN |
+			       PTE_BLOCK_UXN;
```
Comment (sa-kib) [inline board/renesas/rcar3-xen/rcar3-xen.c:129]: nice to have a comment here describing this range

**Comment by firscity** [comment]
@sa-kib all comments are fixed now, thank you for review

**Comment by LKomaryanskiy** [comment]
@firscity good job!
`Acked-by: Leonid Komarianskyi <leonid_komarianskyi@epam.com>`

**Comment by sa-kib** [comment]
LGTM

**Pull request merged**